engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -181,11 +182,63 @@ Type getType() {
     return eigentype;
   }
 
+  @ExportMessage
+  boolean hasMetaObject() {
+    return true;
+  }
+
+  @ExportMessage
+  Type getMetaObject() {
+    return getType();
+  }
+
+  @ExportMessage
+  Object getMetaParents() {
+    assert supertype != null;
```

Comment:
Shouldn't this instead throw `UnsupportedMessageException`?

Reply:
Situation with `null` supertype shall never happen. As such I put it in `assert`. I'd expect it during testing, but not during (production) runtime.